dallinger/experiment_server/dashboard.py

```diff
@@ -652,6 +660,141 @@ def dashboard_lifecycle():
     )
 
 
+def clean_line_dict(line_dict, log_line_number=None):
+    msg = line_dict["message"]
+    if msg.endswith(("-", '"')) and ("GET " in msg or " POST" in msg):
+        msg = '"'.join(msg.split('"')[1:])
+    msg = Ansi2HTMLConverter().convert(msg)
+    parsed_msg = bs4.BeautifulSoup(msg, "html.parser")
+    msg = f"<html>{parsed_msg.head}{parsed_msg.body}</html>"
+    line_dict["message"] = msg
+    if log_line_number is not None:
+        line_dict["log_line_number"] = log_line_number
+    return line_dict
+
+
+LOG_FILE = get_logger_filename()
+
+
+def live_log():
+    """
+    Route that streams log file updates using SSE
+    """
+
+    def generate():
+        for line in Pygtail(LOG_FILE):
+            try:
+                line_dict = clean_line_dict(json.loads(line))
+                line_dict["original_line"] = line
+                yield f"data:{json.dumps(line_dict)}\n\n"
+            except json.decoder.JSONDecodeError:
+                continue
+
+    return Response(generate(), mimetype="text/event-stream")
+
+
+def log_read_lines(line_range):
```

Comment:
Again, I think it would be good to add docstrings to these functions, I don't find them completely self-explanatory from their names.